src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -721,31 +948,36 @@ namespace nbl::ext::imgui
 					{
 						const auto rangeStorageIndex = IDescriptorSetLayoutBase::CBindingRedirect::storage_range_index_t(i);
 						const auto binding = redirect->getBinding(rangeStorageIndex);
-						const auto requestedBindingIx = descriptorType == IDescriptor::E_TYPE::ET_SAMPLED_IMAGE ? m_creationParams.resources.textures.bindingIx : m_creationParams.resources.samplers.bindingIx;
+						const auto requestedBindingIx = descriptorType == IDescriptor::E_TYPE::ET_SAMPLED_IMAGE ? creationParams.resources.texturesInfo.bindingIx : creationParams.resources.samplersInfo.bindingIx;
 
 						if (binding.data == requestedBindingIx)
 						{
 							const auto count = redirect->getCount(binding);
```

Comment:
why are you linear searching through every binding?

You could just call
https://github.com/Devsh-Graphics-Programming/Nabla/blob/52d719bdbde3c000f2c2c454db4d7298e5fa0237/include/nbl/asset/IDescriptorSetLayout.h#L73
once to find your storage index.

Right now you're iterating through all bindings for no reason to find the one you're after and then using the "slow" lookups (by binding_t not storage_index) which perform `lower_bound` each time

Reply:
done by me